File: 1100/1971D/main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() 
{
    int t;
    cin >> t;
    while (t--) 
    {
        string s;
        cin >> s;
        int res = 0, cnt = 0;

        for (int i = 0; i < (int)s.size() - 1; ++i) 
        {
            if (s[i] == '1' && s[i + 1] == '0')
                res++;
            if (s[i] == '0' && s[i + 1] == '1')
                cnt++;
        }
        cout << res + max(cnt, 1) << "\n";
    }
    
    return 0;
}

File: 1100/1971D/main.java
import java.util.*;
public class main
{
    public static void main(String[]args)
    {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        
        while(t-- > 0)
        {
            String s = sc.next();
            int res = 0, cnt = 0;
			for (int i = 0; i < s.length() - 1; ++i) 
			{
				if (s.charAt(i) == '1' && s.charAt(i + 1) == '0') 
				    res++;
				if (s.charAt(i) == '0' && s.charAt(i + 1) == '1') 
				    cnt++;
			}
				
			System.out.println(res + Math.max(cnt, 1));
        }
    }
}